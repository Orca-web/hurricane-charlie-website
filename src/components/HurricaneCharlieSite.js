import React, { Component } from 'react';
import Header from './Header';
import Content from './Content';
import Footer from './Footer';

class HurricaneCharlieSite extends Component {
  constructor() {
    super()
    this.state = { width: 0, gallery: [] };
    this.updateWindowDimensions = this.updateWindowDimensions.bind(this);
  }

  componentDidMount() {
    this.updateWindowDimensions();
    window.addEventListener('resize', this.updateWindowDimensions);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindowDimensions);
  }

  updateWindowDimensions() {
    this.setState({ width: window.innerWidth })
  }

  render() {
    return (
      <div>
        <Header width={this.state.width}/>
        <Content />
        <Footer />
      </div>
    ) 
  }
}

export default HurricaneCharlieSite