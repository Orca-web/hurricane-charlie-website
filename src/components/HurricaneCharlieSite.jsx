import React, { Component } from 'react';
import Header from './Header.jsx';
import Gallery from './Gallery.jsx';
import Footer from './Footer.jsx';

class HurricaneCharlieSite extends Component {
  constructor(props) {
    super(props)
    this.state = {width: 0}
    this.updateWindowDimensions = this.updateWindowDimensions.bind(this);
  }

  componentDidMount() {
    this.updateWindowDimensions();
    window.addEventListener('resize', this.updateWindowDimensions);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindowDimensions);
  }

  updateWindowDimensions() {
    this.setState({ width: window.innerWidth })
  }

  render() {
    return (
      <div>
        <Header width={this.state.width}/>
        <Gallery />
        <Footer />
      </div>
    ) 
  }
}

export default HurricaneCharlieSite