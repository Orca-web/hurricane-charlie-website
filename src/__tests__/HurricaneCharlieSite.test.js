import React from 'react';
import { shallow, configure } from 'enzyme';
import HurricaneCharlieSite from '../components/HurricaneCharlieSite';

const app = shallow(<HurricaneCharlieSite />);

describe('Site', () => {
  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });
  
  it('initializes the `state` with an empty gallery', () => {
    expect(app.state().gallery).toEqual([]);
  });
  
  it('initializes the `state` width to match screen size', () => {
    expect(app.state().width).toEqual(window.innerWidth);
  });
  
  it('window resize edits `state` and matches size', () => {
    global.innerWidth = 500;
    global.dispatchEvent(new Event('resize'));
    expect(app.state().width).toEqual(window.innerWidth);
  });
});