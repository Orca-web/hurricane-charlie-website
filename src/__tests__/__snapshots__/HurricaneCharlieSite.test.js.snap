// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HurricaneCharlieSite />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          width={1024}
        />,
        <Content />,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "width": 1024,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            width={1024}
          />,
          <Content />,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "width": 1024,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
